core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -272,9 +285,11 @@ public CompletableFuture<Map<TopicIdPartition, ShareAcknowledgeResponseData.Part
     ) {
         log.trace("Acknowledge request for topicIdPartitions: {} with groupId: {}",
             acknowledgeTopics.keySet(), groupId);
-        this.shareGroupMetrics.shareAcknowledgement();
+        brokerTopicStats.allTopicsStats().totalShareAcknowledgementRequestRate().mark();
         Map<TopicIdPartition, CompletableFuture<Throwable>> futures = new HashMap<>();
         acknowledgeTopics.forEach((topicIdPartition, acknowledgePartitionBatches) -> {
+            // Update share acknowledgement metrics.
+            brokerTopicStats.topicStats(topicIdPartition.topicPartition().topic()).totalShareAcknowledgementRequestRate().mark();
```

Comment:
Can't you just use `topicIdPartition.topic()`?

Reply:
Done.